lib/Gedmo/Sortable/Mapping/Event/Adapter/ORM.php

```diff
@@ -39,12 +39,38 @@ private function addGroupWhere(QueryBuilder $qb, $groups)
                 $qb->andWhere($qb->expr()->isNull('n.'.$group));
             } else {
                 $qb->andWhere('n.'.$group.' = :group__'.$i);
-                $qb->setParameter('group__'.$i, $value);
+                $qb->setParameter('group__'.$i, $this->getGroupValue($value), $this->getGroupType($value));
             }
             $i++;
         }
     }
 
+    /**
+     * @param $value
+     * @return mixed
+     */
+    private function getGroupValue($value)
+    {
+        if (!is_object($value)) {
+            return $value;
+        }
+        return $this->getObjectManager()->getUnitOfWork()->getSingleIdentifierValue($value);
+    }
+
+    /**
+     * @param $value
+     * @return \Doctrine\DBAL\Types\Type|null|string
+     */
+    private function getGroupType($value)
```

Comment:
`$group` should be passed as well, and the metadata should be fetched like this:
- if the object is a simple value object, `$metaData->getTypeOfField($group)`
- if the object is an entity, your code is ok